cmd/manager/main.go

```diff
@@ -105,100 +137,115 @@ func main() {
 	mgr, err := manager.New(cfg, manager.Options{
 		Namespace:          "",
 		MetricsBindAddress: "0",
+		Scheme:             scheme,
 	})
 	if err != nil {
-		logrus.Fatal(err)
-	}
-
-	logrus.Info("Registering Components.")
-	logrus.Info("setting up scheme")
-	// Setup Scheme for all defined resources
-	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
-		logrus.Fatal(err)
+		logger.Fatal(err)
 	}
-	// Add external resource to scheme
-	if err := olmv1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
-		logrus.Fatal(err)
-	}
-	if err := v1beta1.AddToScheme(mgr.GetScheme()); err != nil {
-		logrus.Fatal(err)
-	}
-	// If the config API is available add the config resources to the scheme
-	if configv1.IsAPIAvailable() {
-		if err := apiconfigv1.AddToScheme(mgr.GetScheme()); err != nil {
-			logrus.Fatal(err)
-		}
-	}
-
-	stopCh := signals.Context().Done()
 
-	var statusReporter status.Reporter = &status.NoOpReporter{}
-	if clusterOperatorName != "" {
-		statusReporter, err = status.NewReporter(cfg, mgr, namespace, clusterOperatorName, os.Getenv("RELEASE_VERSION"), stopCh)
-		if err != nil {
-			logrus.Fatal(err)
-		}
-	}
-
-	// Populate the global default OperatorSources definition and config
-	err = defaults.PopulateGlobals()
-	if err != nil {
-		logrus.Fatal(err)
-	}
-
-	// Setup all Controllers
-	if err := controller.AddToManager(mgr, options.ControllerOptions{}); err != nil {
-		logrus.Fatal(err)
-	}
-
-	logrus.Info("Setting up health checks")
+	logger.Info("setting up health checks")
 	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 	})
 	go http.ListenAndServe(":8080", nil)
 
-	// Wait until this instance becomes the leader.
-	logrus.Info("Waiting to become leader.")
-	err = leader.Become(context.TODO(), leaderElectionConfigMapName)
-	if err != nil {
-		logrus.Fatal(err, "Failed to retry for leader lock")
-	}
-	logrus.Info("Elected leader.")
+	ctx := signals.Context()
+	stopCh := ctx.Done()
+
+	leaderCtx, leaderCancel := context.WithCancel(context.Background())
```

Comment:
why not parent `leaderCtx` w/ the stop signal context: `leaderCtx, leaderCancel := context.WithCancel(ctx)`?